modules/DNN/Perceptron.cpp

```diff
@@ -43,15 +43,23 @@ namespace DirectGraphicalModels {
 			Mat error = gt - solution;
 			for (int i = 0; i < error.rows; i++)
 				error.at<float>(i, 0) *= m_vpNeuronLayers[2]->getActivationFunctionDeriateve()(solution.at<float>(i, 0));
```

Comment:
I think here should be the last(output) layer instead of the third layer:
`m_vpNeuronLayers[2]` --> `m_vpNeuronLayers.back()`

Reply:
Good point, I totally missed that. Will fix it asap.